src/docfx/build/xref/XrefMap.cs

```diff
@@ -3,29 +3,36 @@
 
 using System.Collections.Generic;
 using System.IO;
+using System.Threading.Tasks;
 
 namespace Microsoft.Docs.Build
 {
-    public class XrefMap
+    internal class XrefMap
     {
         // TODO: key could be uid+moniker+locale
-        private readonly IReadOnlyDictionary<string, XrefSpec> _map;
+        private readonly InternalXrefMap _internalXrefMap;
+        private readonly IReadOnlyDictionary<string, XrefSpec> _externalXrefMap;
 
-        private XrefMap(IReadOnlyDictionary<string, XrefSpec> map)
+        private XrefMap(IReadOnlyDictionary<string, XrefSpec> map, InternalXrefMap internalXrefMap)
         {
-            _map = map;
+            _externalXrefMap = map;
+            _internalXrefMap = internalXrefMap;
         }
 
         public XrefSpec Resolve(string uid)
         {
-            if (_map.TryGetValue(uid, out var xrefSpec))
+            if (_internalXrefMap != null && _internalXrefMap.Resolve(uid, out var xrefSpec))
+            {
+                return xrefSpec;
+            }
+            if (_externalXrefMap.TryGetValue(uid, out xrefSpec))
             {
                 return xrefSpec;
             }
             return null;
         }
 
-        internal static XrefMap Create(Docset docset)
+        public static async Task<XrefMap> Create(Context context, Docset docset)
```

Comment:
>Create(Context context, [](start = 42, length = 23)

when and where to push the updated xref files?

Reply:
The internal xref map would be xrefmap.yml for OPS to handle